torch/distributed/tensor/_dispatch.py

```diff
@@ -164,7 +165,8 @@ def dispatch(
         assert output_sharding is not None, "output sharding should not be None"
 
         mesh = op_info.compute_mesh
-        if mesh.get_coordinate() is not None:
+        participating = mesh.get_coordinate() is not None
```

Comment:
When is this false? Is this due to h even sharding leaving one rank 'empty'?

Reply:
hmm. this is kinda news to me.

i would like to see what example this happened in to understand it better.  can you link to it?

usually, we partition the world gpus into various submeshes, but each 'op' still happens on each gpu.  and each submesh dim is an orthogonal partitioning of the world mesh.  Maybe just in some unit tests there are like 8gpus but the test case only initializes 4 of them?